virttest/utils_time.py

```diff
@@ -0,0 +1,229 @@
+    elif params.get("os_type") == "windows":
+        guest_tzs = []
+        guest_tz = []
+        tz_list = session.cmd_output(params["list_tz_cmd"])
+        tz_list = tz_list.splitlines()
+        for line in tz_list:
+            if not line:
+                continue
+            og = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line,
+                           re.M | re.I)
+            if og and og.group(1):
+                og_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2', og.group(1))
+                guest_tz.append(og_)
+                continue
+            elif og:
+                guest_tz.append("+0000")
+                continue
+            og_value = re.search(r'(\S+(?:\s\S+)*)', line, re.M | re.I)
+            if og_value:
+                guest_tz.append(og_value.group(0))
+            guest_tzs.append(guest_tz)
+            guest_tz = []
+
+        for value in guest_tzs:
+            if value[0] == zonecode:
+                return (True, session)
+
+        # Windows guest's timezone not match host's timezone
+        # get the timezone value for windows guest
+        tz_value = ""
+        for value in guest_tzs:
+            if zonecode == value[0]:
+                tz_value = value[1]
+                break
+        status = session.cmd_status(params["set_tz_cmd"] % tz_value)
+
+    if not status:
+        session = vm.reboot(session=session, timeout=600)
+        return (True, session)
+    else:
+        logging.warn("Fail to reboot guest after changing clocksource.")
+        return (False, session)
+
+
+def set_guest_clocksource(session, params, list_guest, source):
+    """
+    Guest's clocksource can be changed dynamically.
+
+    :param session: VM session.
+    :param params: Dictionary with the test parameters.
+    :param list_guest: Avaialble clocksource list for guest.
+    :param source: clocksource you want to set guest to.
+
+    :return: True if set clocksource successful.
+    """
+    try:
+        if list_guest and source in list_guest:
+            if params["os_type"] == "Linux" and session.cmd_status(
+                    "echo %s > %s" % (source, params["clocksource_dir"])):
+                return True
+            else:
+                logging.warn("Can not set clocksource for windows guest.")
+                return False
+        else:
+            logging.warn("Expected clocksource not supported by guest.")
+            return False
+    except Exception:
+        return False
+
+
+def get_guest_clocksource(session, params, list_guest):
+    """
+    Get current clocksource in use.
+
+    :param session: VM session.
+    :param params: Dictionary with the test parameters.
+    :param list_guest: Avaialble clocksource list of guest.
+    :param source: Clocksource want to set guest to.
+
+    :return: clocksource you get.
+    """
+    try:
+        cs = session.cmd_output("cat %s" % params["clocksource_dir"])
+        if cs in list_guest:
```

Comment:
This func is not well defined, `if cs not in list_guest`, this func has no return value. 
Please reorganize it.